.github/workflows/main.yml

```diff
@@ -384,14 +382,16 @@ jobs:
 
       - name: Prepare - Install tools
         run: |
-          brew install automake autoconf-archive libtool ninja
+          brew install --quiet automake autoconf-archive libtool ninja
 
-      - name: Export GitHub Actions cache environment variables
-        uses: actions/github-script@v7
+      - name: Restore vcpkg cache
+        uses: actions/cache/restore@v4
+        id: restore-vcpkg-cache
         with:
-          script: |
-            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
-            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '');
+          path: ~/.cache/vcpkg/archives
+          key: vcpkg-cache-${{ runner.os }}-${{ matrix.arch }}-${{ hashFiles('vcpkg-custom/**') }}
```

Comment:
I would use `key: vcpkg-cache-${{ runner.os }}-${{ matrix.arch }}`

Reply:
See my earlier comment about GitHub Action caches being write once, so the key needs to change if you want to write a new one.

Reply:
With the current setup, if vcpkg-custom doesn't change, it will use the existing cache and only write a new one if the cache didn't exist (or not for the same hash), and if it does change, it will use the existing cache and then write the new cache, which it will use for subsequent actions as well.

Reply:
This is IMO wrong and will not update cache when vcpkg submodule is updated.

Reply:
Then that would be an indication that vcpkg needs to be included in the hash as well.

Reply:
Alternatively, the hash needs to be generated over `~/.cache/vcpkg/archives`. I'll try that as well.

Reply:
OK, that doesn't work, because `hashFiles` can only hash content within the workspace directory. That would mean moving the cache into a location in the workspace and using `VCPKG_DEFAULT_BINARY_CACHE` to specify that location. I'll first try with hashing over vcpkg and vcpkg-custom, before I try and do that.

Reply:
As an aside, do we really need that vcpkg submodule? Can't we use [brew to install vcpkg](https://formulae.brew.sh/formula/vcpkg)?

Reply:
Or am I missing what that submodule actually is?

Reply:
submodule is the recommended approach